### ISSUE TITLE
Load Image Exception Jpg ImageSharp.Formats.Jpg.MissingFF00Exception

### ISSUE THREAD

**Issue by vinhhrv** [SixLabors/ImageSharp#159]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of ImageSharp
- [x] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [x] I have searched [open](https://github.com/JimBobSquarePants/ImageSharp/issuess) and [closed](https://github.com/JimBobSquarePants/ImageSharp/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

Load Image Error

![jpg_bug](https://cloud.githubusercontent.com/assets/769541/24437694/c78c32ec-146c-11e7-9712-de85d73bd467.jpg)

ImageSharp.Formats.Jpg.MissingFF00Exception

### Steps to Reproduce

`using (var file = File.OpenRead("files/jpg_bug.jpg"))
 { 
      var image = Image.Load(file);
}`

Attach file

### System Configuration

- ImageSharp version: 1.0.0-alpha5-00046
- Other ImageSharp packages and versions:
- Environment (Operating system, version and so on): win10
- .NET Framework version: netcoreapp
- Additional information:

**Comment by HelmuthWcs** [comment]
I'm getting the same error on some pictures taken with the samsung gear 360. Failed with alpha2 so I updated, and still failing with latest alpha5
![4](https://cloud.githubusercontent.com/assets/2522424/24452199/608beeb4-1482-11e7-96d7-c07b14925182.jpg)

**Comment by JimBobSquarePants** [comment]
@vinhhrv @HelmuthWcs 

The causes that throw that exception are different in both images. 

The first image is actually correct and highlight a bug in the decoder.

The seconds image however is buggy and will require a workaround. Here's the output from JpegSnoop

```
*** Decoding SCAN Data ***
  OFFSET: 0x0000D03B
  Scan Decode Mode: Full IDCT (AC + DC)

  Scan Data encountered marker   0xFFD9 @ 0x00785631.0
*** ERROR: Can't find huffman bitstring @ 0x0078562F.1, table 1, value [0xffffb200]
*** ERROR: Bad huffman code @ 0x0078562F.0
*** ERROR: Bad scan data in MCU(449,224): Chr(Cr) CSS(0,0) @ Offset 0x0078562F.1
           MCU located at pixel=(7184,3584)

```

**Comment by brianpopow** [comment]
unfortunately i am still getting this error: MissingFF00Exception. 

The image is the following:

![25246205](https://user-images.githubusercontent.com/38701097/40072449-361932d6-5874-11e8-8c13-00e9395d2c10.jpg)

It occurs in beta3 and in the current master branch. 

I have tested the image with JpegSnoop and imagemagick identify, both do not report any issue with the image.

**Comment by antonfirsov** [comment]
@brianpopow should be fixed as soon as we merge #571 (very soon!)

**Comment by brianpopow** [comment]
nice, i am glad to hear that! 

I have tested it with the swap-jpeg-decoder branch and it did indeed fix the issue.
